src/components/judging/QualifierJudging.vue

```diff
@@ -0,0 +1,251 @@
+
+@Component({
+    props: {
+        round: {
+            type: Object,
+            required: true,
+        },
+        criterias: {
+            type: Array,
+            required: true,
+        },
+        judgingDone: {
+            type: Array,
+            required: true,
+        },
+    },
+    components: {
+        PageHeader,
+    },
+})
+
+export default class Qualifier extends Vue {
+
+    judgingDone!: QualifierJudging[];
+    round!: Round;
+    editingSubmission: Submission | null = null;
+    editingCriteria: Criteria | null = null;
+    editingScore = 0;
+    editingComment = '';
+    alertInfo = '';
+    alertSuccess!: boolean;
+
+    selectForEditing (submission: Submission, criteria: Criteria): void {
+        this.editingSubmission = submission;
+        this.editingCriteria = criteria;
+        this.editingScore = 0;
+        this.editingComment = '';
+        this.alertInfo = '';
+        this.alertSuccess = false;
+
+        const judgingToCriterias = this.getJudgingToCriterias(submission.id, criteria.id);
+
+        if (judgingToCriterias) {
+            this.editingComment = judgingToCriterias.comment;
+            this.editingScore = judgingToCriterias.score;
+        }
+    }
+
+    getJudgingToCriterias(submissionId: number, criteriaId: number): QualifierJudgingToCriterias | null {
+        const judging = this.judgingDone.find(j => j.submissionId === submissionId);
+        if (!judging)
+            return null;
+
+        const qualifierJudgingToCriterias = judging.qualifierJudgingToCriterias.find((q) => q.criteriaId === criteriaId);
+        if (!qualifierJudgingToCriterias)
+            return null;
+
+        return qualifierJudgingToCriterias;
+    }
+
+    getScore(submissionId: number, criteriaId: number): number {
+        const qualifierJudgingToCriterias = this.getJudgingToCriterias(submissionId, criteriaId);
+        if (!qualifierJudgingToCriterias)
+            return 0;
+
+        return qualifierJudgingToCriterias.score;
+    }
+
+    isCompleted(submissionId: number): boolean {
+        const judging = this.judgingDone.find(j => j.submissionId === submissionId);
+        if (!judging)
+            return false;
+
+        return judging.qualifierJudgingToCriterias.length === 4;
+    }
+
+    async save (): Promise<void> {
+        this.alertInfo = '';
+        const res = await Axios.post('/api/judging/save', {
+            submissionId: this.editingSubmission?.id,
+            criteriaId: this.editingCriteria?.id,
+            score: this.editingScore,
+            comment: this.editingComment,
+            round: this.round,
+        });
+
+        if (!res.data)
+            return;
+
+        const info = res.data.success || res.data.error;
+        this.alertInfo = info;
+        this.alertSuccess = res.data.success != undefined;
+
+        if (!this.alertSuccess)
+            return;
+
+        const { judgingDone } = res.data;
+        this.judgingDone = judgingDone;
```

Comment:
`this.$emit('update:judging-done', judgingDone)`
https://vuejs.org/v2/guide/components-custom-events.html